skyportal/tests/conftest.py

```diff
@@ -1638,3 +1645,48 @@ def analysis_token(user):
     )
     yield token_id
     delete_token(token_id)
+
+
+@pytest.fixture(scope="function")
+def gcn_GW190425():
+    datafile = f'{os.path.dirname(__file__)}/data/GW190425_initial.xml'
+
+    with open(datafile, 'rb') as fid:
+        payload = fid.read()
+
+    user_id = 1
+    with DBSession() as session:
+        event_id = post_gcnevent_from_xml(payload, user_id, session)
+        time.sleep(30)
+        yield event_id
+        GcnEventFactory.teardown(event_id)
```

Comment:
I'm not sure how good it is to keep the session open until teardown...
If you are passing an ID it would probably be ok to just close the session, yield the ID, and open a new one to delete the data.